src/aiida/transports/plugins/ssh_async.py

```diff
@@ -764,6 +764,110 @@ async def copytree_async(
         """
         return await self.copy_async(remotesource, remotedestination, dereference, recursive=True, preserve=preserve)
 
+    async def compress_async(
+        self, format, remotesource, remotedestination, rootdir, overwrite=True, dereference=False, *args, **kwargs
+    ):
+        """Compress a remote directory.
+        This method should be able to handle remotedir with glob patterns.
+
+        :param format: format of compression, should support: 'tar', 'tar.gz', 'tar.bz', 'tar.xz'
+        :param remotesource: path to the remote directory(or file) to compress
+        :param remotedestination: path to the remote destination file (inclusing file name).
+        :param rootdir: the path that compressed files are be relative to.
```

Comment:
I am not sure about rootdir, since you have remotesource  which could be just a list of absolute paths, then rootdir is not needed. Sure it gives extra flexibility but also extra complexity that need to be considered.